src/phase.cs

```diff
@@ -0,0 +1,1299 @@
+	[DisableDefaultCtor]
+	interface PhaseEnvelopeDistanceModelParameters
+	{
+		[Export ("initWithEnvelope:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (PhaseEnvelope envelope);
+
+		[Export ("envelope", ArgumentSemantic.Strong)]
+		PhaseEnvelope Envelope { get; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASEDucker")]
+	[DisableDefaultCtor]
+	interface PhaseDucker
+	{
+		[Export ("initWithEngine:sourceGroups:targetGroups:gain:attackTime:releaseTime:attackCurve:releaseCurve:")]
+		IntPtr Constructor (PhaseEngine engine, NSSet<PhaseGroup> sourceGroups, NSSet<PhaseGroup> targetGroups, double gain, double attackTime, double releaseTime, PhaseCurveType attackCurve, PhaseCurveType releaseCurve);
+
+		[Export ("activate")]
+		void Activate ();
+
+		[Export ("deactivate")]
+		void Deactivate ();
+
+		[Export ("isActive")]
+		bool IsActive { get; }
+
+		[Export ("sourceGroups", ArgumentSemantic.Copy)]
+		NSSet<PhaseGroup> SourceGroups { get; }
+
+		[Export ("targetGroups", ArgumentSemantic.Copy)]
+		NSSet<PhaseGroup> TargetGroups { get; }
+
+		[Export ("gain")]
+		double Gain { get; }
+
+		[Export ("attackTime")]
+		double AttackTime { get; }
+
+		[Export ("releaseTime")]
+		double ReleaseTime { get; }
+
+		[Export ("attackCurve")]
+		PhaseCurveType AttackCurve { get; }
+
+		[Export ("releaseCurve")]
+		PhaseCurveType ReleaseCurve { get; }
+
+		[Export ("identifier")]
+		string Identifier { get; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASEGroupPresetSetting")]
+	[DisableDefaultCtor]
+	interface PhaseGroupPresetSetting
+	{
+		[Export ("initWithGain:rate:gainCurveType:rateCurveType:")]
+		IntPtr Constructor (double gain, double rate, PhaseCurveType gainCurveType, PhaseCurveType rateCurveType);
+
+		[Export ("gain")]
+		double Gain { get; }
+
+		[Export ("rate")]
+		double Rate { get; }
+
+		[Export ("gainCurveType")]
+		PhaseCurveType GainCurveType { get; }
+
+		[Export ("rateCurveType")]
+		PhaseCurveType RateCurveType { get; }
+	}
+
+	[NoWatch, NoTV, Mac (12,0), iOS (15,0), MacCatalyst (15,0)]
+	[BaseType (typeof (NSObject), Name="PHASEGroupPreset")]
+	[DisableDefaultCtor]
+	interface PhaseGroupPreset
+	{
+		[Export ("initWithEngine:settings:timeToTarget:timeToReset:")]
+		[DesignatedInitializer]
+		IntPtr Constructor (PhaseEngine engine, NSDictionary<NSString, PhaseGroupPresetSetting> settings, double timeToTarget, double timeToReset);
+
+		[Export ("settings")]
+		NSDictionary<NSString, PhaseGroupPresetSetting> Settings { get; }
+
+		[Export ("timeToTarget")]
+		double TimeToTarget { get; }
+
+		[Export ("timeToReset")]
+		double TimeToReset { get; }
+
+		[Export ("activate")]
+		void Activate ();
+
+		[Export ("activateWithTimeToTargetOverride:")]
+		void ActivateWithTimeToTargetOverride (double timeToTargetOverride);
```

Comment:
```suggestion
		void Activate (double timeToTargetOverride);
```
Maybe this is sufficient, looks like this is the Swift name as well